api.js

```diff
@@ -77,8 +77,10 @@ module.exports = function attachAPI(app, {io, db}) {
 
     // Extra details for a channel - these aren't returned in the channel list API,
     // but are when a specific channel is fetched.
-    channelDetail: c => Object.assign(serialize.channelShort(c), {
-      pinnedMessageIDs: c.pinnedMessageIDs
+    channelDetail: async c => Object.assign(serialize.channelShort(c), {
+      pinnedMessages: await Promise.all(c.pinnedMessageIDs.map(
+        async id => serialize.message(await db.messages.findOne({_id: id}))
```

Comment:
...what happens if the database method resolves with `null`?

Reply:
Good news for people who love bad news :tada:

Should it just filter out null messages? - The only case I can think of where a message is null is if it's been deleted. That's not in the API yet, but it might be in the future (though I'm not too certain how to go about that). Still good to be careful.

Reply:
Yeah, I think its worth checking for deleted messages at this point. Deleted as in `null` 📦

Reply:
Fixed.